tests/rptest/tests/enterprise_features_license_test.py

```diff
@@ -76,42 +76,51 @@ def _put_license(self):
         assert self.admin.put_license(license).status_code == 200, \
             "License update failed"
 
-        def obtain_license(node):
+        def obtain_configured_license(node):
             lic = self.admin.get_license(node=node)
-            return (lic is not None and lic['loaded'] is True, lic)
+            return (self.admin.is_sample_license(lic), lic)
 
         result = None
         for n in self.redpanda.nodes:
-            resp = wait_until_result(lambda: obtain_license(n),
+            resp = wait_until_result(lambda: obtain_configured_license(n),
                                      timeout_sec=5,
                                      backoff_sec=1)
             assert resp['license'] is not None, "License upload failed!"
             result = resp['license'] if result is None else result
 
         return result
 
+    def _disable_evaluation_period(self):
+        self.redpanda.set_environment(
+            {'__REDPANDA_DISABLE_BUILTIN_TRIAL_LICENSE': True})
+        self.redpanda.restart_nodes(self.redpanda.nodes)
```

Comment:
suggestion: maybe move this to `redpanda.py`, but this isn't a blocker

Reply:
FWIW the restart is only needed if the brokers have already been started. Tests that manually start the brokers (empty `setUp` method, eg. the tests in `license_enforcement_test.py`) can set the env var before the first broker is started and the brokers will pick it up on startup.

In this test case, I chose to restart the nodes instead of switching to manually start them because that seemed easier.

Reply:
this is all getting ripped up in my PR anyway. i think "get the test passing and move on" is the right course of action.